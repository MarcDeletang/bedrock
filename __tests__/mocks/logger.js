const importMock = require('../../app/mocks/logger.ts');
const { mockLogger, log } = importMock;

const request = {};
const tags = ['hello:)'];
const message = 'world :D';

// Those tests should never fail, even with
// console activated (returns undefined)
describe('Testing the logger', () => {
  it('Check the logger properties', () => {
    expect(mockLogger).toMatchSnapshot();
  });

  it('Check the logger server method', () => {
    expect(mockLogger.server(tags, message)).toMatchSnapshot();
  });

  it('Check the logger request method', () => {
    expect(mockLogger.request(request, tags, message)).toMatchSnapshot();
  });

  it('Check the log method (coverage)', () => {
    expect(log(tags, message, null, true)).toMatchSnapshot();
  });

});
