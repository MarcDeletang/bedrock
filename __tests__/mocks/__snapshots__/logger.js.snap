// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the logger Check the log method (coverage) 1`] = `undefined`;

exports[`Testing the logger Check the logger properties 1`] = `MockLogger {}`;

exports[`Testing the logger Check the logger request method 1`] = `undefined`;

exports[`Testing the logger Check the logger server method 1`] = `undefined`;
